actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -414,6 +414,20 @@ func (pb *Actor) Cancel(vmctx exec.VMContext, chid *types.ChannelID) (uint8, err
 			return errors.NewFaultError("Expected PaymentChannel from channels lookup")
 		}
 
+		if channel.Redeemed {
+			if channel.Condition == nil {
+				return errors.NewFaultError("Channel cannot be cancelled due to successful redeem")
+			} else {
+				err := checkCondition(vmctx, channel, channel.Condition, []interface{}{})
```

Comment:
This is a little confusing. It would be better to extract the part of `checkCondtion` that makes the call from the part that handles storing the condition, and just call that here.